parlai/agents/hugging_face/dict.py

```diff
@@ -31,15 +33,30 @@ class HuggingFaceDictionaryAgent(DictionaryAgent, ABC):
     Use Hugging Face tokenizers.
     """
 
-    def __init__(self, opt):
-        super().__init__(opt)
-        # initialize from vocab path
-        self.tokenizer = self.get_tokenizer(opt)
+    def __init__(self, opt: Opt, shared=None):
+        if not shared:
+            self.hf_tokenizer = self.get_tokenizer(opt)
+            self.tok2ind = self.hf_tokenizer.get_vocab()
+            self.ind2tok = {v: k for k, v in self.tok2ind.items()}
+        else:
+            self.hf_tokenizer = shared['hf_tokenizer']
+            self.tok2ind = shared['tok2ind']
+            self.ind2tok = shared['ind2tok']
+
+        self.freq = defaultdict(int)
+        for tok in self.tok2ind:
+            self.freq[tok] = 1
+        self.minfreq = opt.get('dict_minfreq', DictionaryAgent.default_minfreq)
+
         self.override_special_tokens(opt)
-        for i in range(self.tokenizer.vocab_size):
-            token = self.tokenizer._convert_id_to_token(i)
-            self.add_token(token)
-            self.freq[token] = 1
+        self._unk_token_idx = self.hf_tokenizer.unk_token_id
```

Comment:
Should these lines be swapped? Right now `_unk_token_idx` from `override_special_tokens` is overridden.

Reply:
good catch